src/media-query-provider.js

```diff
@@ -56,13 +56,12 @@ class MediaQueryProvider extends React.Component {
   }
 
   componentWillUnmount() {
-    this.mediaQueryListInstanceMap.forEach((_, mediaQueryList) => {
-      mediaQueryList.removeListener(this.mediaQueryListener);
-    });
+    this.mediaQueryListInstanceMap.forEach(mediaQueryList =>
```

Comment:
I think this won't work, you need the second arg for the value so will have to keep the `(_, mediaQueryList)`

Reply:
Ah my bad, nice one